packages/modular-scripts/react-scripts/config/parts/esmViewConfig.js

```diff
@@ -68,23 +75,28 @@ function createExternalRewriter(dependencyMap) {
 }
 
 // Virtual entrypoint if we're starting a ESM view - see https://github.com/webpack/webpack/issues/6437
-function getVirtualTrampoline(paths) {
+function getVirtualTrampoline(paths, useReactCreateRoot) {
   // Build the relative path between the root and the entrypoint.
   const relativeEntrypointPath = path
     .relative(paths.appPath, paths.appIndexJs)
     .split(path.sep)
     .join(path.posix.sep); // Separator could be win32 on Windows system, since it comes from a filesystem path. Force it to be posix since it's an URL
 
-  const entryPointPath = `'./${relativeEntrypointPath}'`;
-  const string = `
-  import ReactDOM from 'react-dom'
-  import React from 'react';
-  import Component from ${entryPointPath};
-  const DOMRoot = document.getElementById('root');
-  ReactDOM.render(React.createElement(Component, null), DOMRoot);
-	`;
+  const entryPointPath = `./${relativeEntrypointPath}`;
+  const content = useReactCreateRoot
+    ? `import { createRoot } from "react-dom/client";
```

Comment:
nit: we could use the tiny `dedent` library to keep this source a bit more readable

alternatively, these two mini templates could exist as static variables outside of this function - it's slightly busy now that the ternerary operator is there

not a big deal though considering it's only about 10 lines

Reply:
Done